src/test/java/WebDriverScenarioWithTests.java

```diff
@@ -0,0 +1,132 @@
+                        return ((JavascriptExecutor) driver).executeScript("return document.readyState").equals("complete");
+                    }
+                };
+        WebDriverWait wait = new WebDriverWait(driver, 10);
+        logger.info("1. Launch Chrome browser in maximized mode.");
+        driver.get("https://google.com");
+        wait.until(pageLoadCondition);
+        driver.manage().window().maximize();
+
+        logger.info("2. Open 'google.com' and search for 'wikipedia'");
+        WebElement searchInput = driver.findElement(By.name("q"));
+        searchInput.sendKeys("wikipedia" + Keys.ENTER);
+        List<WebElement> searchResults = driver.findElements(By.xpath("//cite[contains(text())]"));
+        logger.info(String.format("Search results number for requested text: %s", searchResults.size()));
+        Assert.assertTrue(Boolean.parseBoolean(String.valueOf(searchResults.contains("wikipedia"))), "Search results do not contain expected text.");
+
+        logger.info("3. Validate that first found link is related to wikipedia."); //contains()
+        WebElement firstElementOfSearchResults = driver.findElement(By.xpath("//cite[@class='iUh30 Zu0yb qLRx3b tjvcx']"));
+        String contentOfFirstSearchResult = firstElementOfSearchResults.getText();
+        new WebDriverWait(driver, 50).until(ExpectedConditions
+                .presenceOfAllElementsLocatedBy(By.xpath("//cite[contains(text(),'wikipedia')]")));
+        logger.info(String.format("First element of search results is %s", contentOfFirstSearchResult));
+        wait.until(pageLoadCondition);
+        Assert.assertTrue(contentOfFirstSearchResult.contains("wikipedia"), "First element of search results do not contain expected text.");
+
+        logger.info("4. Navigate to https://en.wikipedia.org via first found link.");
+        WebElement firstEnglishWikipediaLink = driver.findElement(By.xpath("//cite[text()='https://en.wikipedia.org']"));
+        new WebDriverWait(driver, 20).until(ExpectedConditions
+                .presenceOfAllElementsLocatedBy(By.xpath("//cite[text()='https://en.wikipedia.org']")));
+        String contentOfFirstEnglishWikipediaLink = firstEnglishWikipediaLink.getText();
+        firstEnglishWikipediaLink.click();
+        wait.until(pageLoadCondition);
+
+        logger.info("5. Assert that we are currently on wikipedia page.");
+        logger.info(String.format("The link clicked is %s", contentOfFirstEnglishWikipediaLink));
+
+        Assert.assertTrue(contentOfFirstEnglishWikipediaLink.contains("wikipedia"), "The element is clicked do not contains expected text.");
+
+        logger.info("6. Get Page Title name and Title length -> Print Page Title and Title length on the Console via logger.");
+        String titleOfClickedEnglishWikipediaLink = driver.getTitle();
+        int titleLength = titleOfClickedEnglishWikipediaLink.length();
+        logger.info("The title of clicked link is " + titleOfClickedEnglishWikipediaLink
```

Comment:
Please use String.format to avoid strings concatenation

Reply:
Refactored, thank you.
Please see in the following commit.

Code is a bit same, from my point of view... Is it common rule to use NOT concatenation?

Reply:
no, it is not common rule. Just a suggestion. We can go and try over different variants of work with String.

Reply:
I see, thank you for clarifying and sorry for stupid questions...)